messenger.rst

```diff
@@ -654,35 +736,181 @@ This middleware can be configured to define the entity manager to use:
         // config/packages/messenger.php
         $container->loadFromExtension('framework', [
             'messenger' => [
-                'buses' => [
-                    'command_bus' => [
-                        'middleware' => [
-                            // Using the default configured entity manager name
-                            'doctrine_transaction',
-                            // Using another entity manager
-                            ['id' => 'doctrine_transaction', 'arguments' => ['custom']],
-                        ],
+                'transports' => [
+                    'async_priority_high' => 'dsn' => '%env(MESSENGER_TRANSPORT_DSN)%?queue_name=high_priority',
+                    'async_priority_low' => [
+                        'dsn' => '%env(MESSENGER_TRANSPORT_DSN)%',
+                        'options' => [
+                            'queue_name' => 'normal_priority'
+                        ]
                     ],
                 ],
             ],
         ]);
 
-Defining such configurable middleware is based on Symfony's
-:doc:`dependency injection </service_container>` features:
+Options defined under ``options`` take precedence over ones defined in the DSN.
+
+==================  =================================== =======
+     Option               Description                   Default
+==================  =================================== =======
+table_name          Name of the table                   messenger_messages
+queue_name          Name of the queue (a column in the  default
+                    table, to use-use one table for
+                    multiple transports)
+redeliver_timeout   Timeout before retrying a messages  3600
+                    that's in the queue but in the
+                    "handling" state (if a worker died
+                    for some reason, this will occur,
+                    eventually you should retry the
+                    message)
+auto_setup          Whether the table should be created
+                    automatically during send / get.    true
+==================  =================================== =======
+
+redis Transport
+~~~~~~~~~~~~~~~
+
+.. versionadded:: 4.3
+
+    The Redis transport was introduced in Symfony 4.3.
+
+The Redis transport uses `streams`_ to queue messages.
+
+.. code-block:: bash
+
+    # .env
+    MESSENGER_TRANSPORT_DSN=redis://localhost:6379/messages
+
+To use the Redis transport, you will need the Redis PHP extension (^4.2) and
+a running Redis server (^5.0).
+
+.. caution::
+
+    The Redis transport does not support "delayed" messages.
+
+A number of options can be configured via the DSN of via the ``options`` key
+under the transport in ``messenger.yaml``:
+
+==================  =================================== =======
+     Option               Description                   Default
+==================  =================================== =======
```

Comment:
I think we should fix these outline lines to match the longest length of this column

Reply:
I kind of did it on purpose :). These "simple" tables are meant to be easier to write - I think we should be a bit flexible on them.

Reply:
Hmm, I guess it's just me that goes "whaaaa" when seeing outlining like this :)